src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,48 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 2;
+    private final int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length < THRESHOLD) {
+            return;
+        }
+        int mid = array.length / 2;
+
+        int[] left = new int[mid];
+        System.arraycopy(array, 0, left, 0, mid);
+
+        int[] right = new int[array.length - mid];
+        System.arraycopy(array, mid, right, 0, array.length - mid);
+
+        invokeAll(new MergeSortAction(left), new MergeSortAction(right));
+        merge(left, right);
+    }
 
+    private void merge(int[] left, int[] right) {
+        int i = 0;
+        int j = 0;
+        int k = 0;
+        while (i < left.length && j < right.length) {
+            if (left[i] < right[j]) {
+                array[k++] = left[i++];
+            } else {
+                array[k++] = right[j++];
```

Comment:
You can avoid using k variable
```suggestion
                array[i + j] = left[i++];
                i++
            } else {
                array[i + j] = right[j++];
                j++
```